tests/integration/ha_tests/helpers.py

```diff
@@ -77,15 +81,20 @@ async def are_all_db_processes_down(ops_test: OpsTest, process: str) -> bool:
         pgrep_cmd = ("pgrep", "-x", process)
 
     try:
-        for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(3)):
+        for attempt in Retrying(stop=stop_after_delay(400), wait=wait_fixed(3)):
             with attempt:
                 for unit in ops_test.model.applications[app].units:
-                    _, raw_pid, _ = await ops_test.juju(
-                        "ssh", "--container", "postgresql", unit.name, *pgrep_cmd
+                    pod_name = unit.name.replace("/", "-")
+                    call = subprocess.run(
+                        f"{KUBECTL} -n {ops_test.model.info.name} exec {pod_name} -c postgresql -- {' '.join(pgrep_cmd)}",
+                        shell=True,
```

Comment:
Juju ssh creates a shadow process that was triggering the check if patroni is still running.